File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace GridViewEditHotKeyMvc.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(Invoice.GetData())
		End Function

		Public Function GridViewAddNewPartial(ByVal model As Invoice) As ActionResult
			' Update the underlying datasource here
			' ...
			Return PartialView("GridViewPartial", Invoice.GetData())
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace GridViewEditHotKeyMvc.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial() {
            return PartialView(Invoice.GetData());
        }

        public ActionResult GridViewAddNewPartial(Invoice model) {
            // Update the underlying datasource here
            // ...
            return PartialView("GridViewPartial", Invoice.GetData());
        }
    }
}
